lib/src/conflicts.rs

```diff
@@ -28,11 +29,29 @@ use crate::merge::{Merge, MergeBuilder, MergedTreeValue};
 use crate::repo_path::RepoPath;
 use crate::store::Store;
 
-const CONFLICT_START_LINE: &[u8] = b"<<<<<<<\n";
-const CONFLICT_END_LINE: &[u8] = b">>>>>>>\n";
-const CONFLICT_DIFF_LINE: &[u8] = b"%%%%%%%\n";
-const CONFLICT_MINUS_LINE: &[u8] = b"-------\n";
-const CONFLICT_PLUS_LINE: &[u8] = b"+++++++\n";
+const CONFLICT_START_LINE: &[u8] = b"<<<<<<<";
+const CONFLICT_END_LINE: &[u8] = b">>>>>>>";
+const CONFLICT_DIFF_LINE: &[u8] = b"%%%%%%%";
+const CONFLICT_MINUS_LINE: &[u8] = b"-------";
+const CONFLICT_PLUS_LINE: &[u8] = b"+++++++";
+const CONFLICT_START_LINE_CHAR: u8 = CONFLICT_START_LINE[0];
+const CONFLICT_END_LINE_CHAR: u8 = CONFLICT_END_LINE[0];
+const CONFLICT_DIFF_LINE_CHAR: u8 = CONFLICT_DIFF_LINE[0];
+const CONFLICT_MINUS_LINE_CHAR: u8 = CONFLICT_MINUS_LINE[0];
+const CONFLICT_PLUS_LINE_CHAR: u8 = CONFLICT_PLUS_LINE[0];
+
+/// A conflict marker is one of the separators, optionally followed by a space
+/// and some text.
+// TODO: All the `{7}` could be replaced with `{7,}` to allow longer
+// separators. This could be useful to make it possible to allow conflict
+// markers inside the text of the conflicts.
+static CONFLICT_MARKER_REGEX: once_cell::sync::Lazy<Regex> = once_cell::sync::Lazy::new(|| {
+    Regex::new(
+        r"(<{7}|>{7}|%{7}|\-{7}|\+{7})( .*)?
+",
+    )
+    .unwrap()
+});
```

Comment:
Maybe this is nitpicky but I don't see why a regex is needed here at all when it's only used as a shorthand to match against a handful of possible outcomes? We could just write a 10 line string handling routine here, right?